examples/oemof_0.2/excel_reader/dispatch.py

```diff
@@ -33,125 +36,303 @@
+
+    edge_color : string
+        Hex color code oder matplotlib color for edge color.
+
+    plot : boolean
+        Show matplotlib plot.
+
+    node_size : integer
+        Size of nodes.
+
+    with_labels : boolean
+        Draw node labels.
+
+    arrows : boolean
+        Draw arrows on directed edges. Works only if an optimization_model has
+        been passed.
+    layout : string
+        networkx graph layout, one of: neato, dot, twopi, circo, fdp, sfdp.
+    """
+    if type(node_color) is dict:
+        node_color = [node_color.get(g, '#AFAFAF') for g in grph.nodes()]
+
+    # set drawing options
+    options = {
+        'prog': 'dot',
+        'with_labels': with_labels,
+        'node_color': node_color,
+        'edge_color': edge_color,
+        'node_size': node_size,
+        'arrows': arrows
+    }
+
+    # draw graph
+    pos = nx.drawing.nx_agraph.graphviz_layout(grph, prog=layout)
+
+    nx.draw(grph, pos=pos, **options)
+
+    # add edge labels for all edges
+    if edge_labels is True and plt:
+        labels = nx.get_edge_attributes(grph, 'weight')
+        nx.draw_networkx_edge_labels(grph, pos=pos, edge_labels=labels)
+
+    # show output
+    if plot is True:
+        plt.show()
 
 
 logger.define_logging()
-datetime_index = pd.date_range(
-    '2030-01-01 00:00:00', '2030-01-14 23:00:00', freq='60min')
+datetime_index = pd.date_range('2016-01-01 00:00:00',
+                               '2016-01-07 23:00:00',
+                               freq='60min')
 
 # model creation and solving
 logging.info('Starting optimization')
 
 # initialisation of the energy system
-es = solph.EnergySystem(timeindex=datetime_index)
+esys = solph.EnergySystem(timeindex=datetime_index)
 
-# adding all nodes and flows to the energy system
-# (data taken from excel-file)
-nodes = nodes_from_excel(
-    os.path.join(os.path.dirname(__file__), 'scenarios.xlsx',))
+# read node data from Excel sheet
+nd = nodes_from_excel(
+    os.path.join(os.path.dirname(__file__), 'scenario.xlsx',))
 
-es.add(*nodes.values())
+# create nodes from Excel sheet data
+nodes = create_nodes(nd=nd)
 
-print("********************************************************")
-print("The following objects has been created from excel sheet:")
-for n in es.nodes:
+# add nodes and flows to energy system
+esys.add(*nodes)
+
+print("*********************************************************")
+print("The following objects have been created from excel sheet:")
+for n in esys.nodes:
     oobj = str(type(n)).replace("<class 'oemof.solph.", "").replace("'>", "")
     print(oobj + ':', n.label)
-print("********************************************************")
+print("*********************************************************")
 
 # creation of a least cost model from the energy system
-om = solph.Model(es)
+om = solph.Model(esys)
 om.receive_duals()
 
 # solving the linear problem using the given solver
 om.solve(solver='cbc')
 
+# plot esys graph
+graph = create_nx_graph(esys)
```

Comment:
Maybe try/except would help to keep it running even if `pygraphviz` is not installed.

```python
graph = create_nx_graph(esys, filename='/home/somebody/my_graph.graphml')
```
This will dump the graph to disc. You can open it with yEd or gephi.